### PR TITLE
[review]ユーザーは商品一覧ページを閲覧できる systemテスト

### PR DESCRIPTION
**Pull request by Takumi-Tajima**
- ユーザーの商品テストを追加
  - 商品の一覧を閲覧できる

## Release Notes

- **New Features**
  - Introduced a responsive user navigation header with login/logout options.
  - Added user registration and login forms for enhanced user authentication.
  - Implemented user-specific routing and item management functionality.
  - Enhanced admin interface with a dedicated layout and clearer navigation.
  - Updated the admin header to clarify the interface for admin users.
  - Added a new layout template for the admin section to provide a consistent structure.

- **Bug Fixes**
  - Removed unnecessary test controller to simplify the codebase.

- **Chores**
  - Set up a factory for user instances to streamline testing.
  - Created system tests to ensure proper functionality for logged-in users accessing item listings.
  - Enhanced CI/CD processes with new linting and testing workflows.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,6 +1,10 @@
 Rails.application.routes.draw do
   devise_for :users
+  scope module: :users do
```
Comment (uekkie) [inline config/routes.rb:3]: user の root が必要ない？

File: spec/system/users/items_spec.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+RSpec.describe '商品管理機能', type: :system do
+  context 'ログインしている時' do
+    let(:user) { create(:user) }
+    let!(:item) { create(:item, name: '大根', price: '200') }
+
+    before do
+      sign_in user
+    end
+
+    it '商品の一覧ページを閲覧できること' do
```
Comment (uekkie) [inline spec/system/users/items_spec.rb:12]: `xit` にしておくと、まだ動かないテストというのはわかる

File: config/routes.rb (hunk 1 above)
Comment (Takumi-Tajima) [reply config/routes.rb:3]: 次のタスクでルート設定します！

File: spec/system/users/items_spec.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+RSpec.describe '商品管理機能', type: :system do
+  context 'ログインしている時' do
+    let(:user) { create(:user) }
+    let!(:item) { create(:item, name: '大根', price: '200') }
+
+    before do
+      sign_in user
+    end
+
+    xit '商品の一覧ページを閲覧できること' do
```
[the remark this answers is not in the document]
Comment (uekkie) [reply spec/system/users/items_spec.rb:12]: たしかに次から理由も書いておくといいかもね

**Review by uekkie — APPROVED** [review]

**Pull request merged**